sparktk-core/src/main/scala/org/trustedanalytics/sparktk/models/SparkTkModelAdapter.scala

```diff
@@ -0,0 +1,149 @@
+import org.trustedanalytics.sparktk.saveload.TkSaveableObject
+
+/**
+ * Wrapper that is able to read a MAR file containing sparktk model and related jars; and then loads and returns the Model for scoring
+ */
+class SparkTkModelAdapter() extends ModelReader {
+  private val logger = LoggerFactory.getLogger(this.getClass)
+  val MODEL_NAME = "modelClassName"
+
+  override def read(modelZipFileInput: File, classLoader: URLClassLoader, jsonMap: Map[String, String]): Model = {
+    val zipInputStream = new ZipInputStream(new FileInputStream(modelZipFileInput))
+    read(zipInputStream, classLoader, jsonMap)
+  }
+
+  /**
+   * Takes a zip input stream with MAR file contents for sparktk models; then loads and returns a sparktk model
+   *
+   * @param modelZipStreamInput stream with MAR file contents for sparktk models
+   * @return loads and returns the sparktk model
+   */
+  override def read(modelZipStreamInput: ZipInputStream, classLoader: URLClassLoader, jsonMap: Map[String, String]): Model = {
+    logger.info("Sparktk model Adapter called")
+    val sparktkObject = classLoader.loadClass(jsonMap(MODEL_NAME) + "$").getField("MODULE$").get(null).asInstanceOf[TkSaveableObject]
+    Thread.currentThread().setContextClassLoader(classLoader)
+    val tc = createSimpleContext(modelZipStreamInput)
+    sparktkObject.load(tc, getModelPath(modelZipStreamInput)).asInstanceOf[Model]
+  }
+
+  /**
+   * Create a TkContext with simple, local-mode SparkContext running with local fs
+   * @return a new TkContext
+   */
+  private def createSimpleContext(modelZipStreamInput: ZipInputStream): TkContext = {
+    val conf = new SparkConf()
+      .setAppName("simple")
+      .setMaster("local[1]")
+      .set("spark.ui.enabled", "false")
+      .set("spark.hadoop.fs.default.name", "file:///")
+    val sc = new SparkContext(conf)
+    new TkContext(new JavaSparkContext(sc))
+  }
+
+  /**
+   * Creates and returns the path to a temporary directory holding the sparktk model
+   * @param modelZipStreamInput Zip input stream containing the sparktk model
+   * @return Location of the temporary directory holding the model
+   */
+  private def getModelPath(modelZipStreamInput: ZipInputStream): String = {
+    var tmpDir: Path = null
+    val bytesIn = new Array[Byte](4096)
+    var fileStr: String = null
+    try {
+      tmpDir = Files.createTempDirectory("sparktk-scoring")
+      var entry = modelZipStreamInput.getNextEntry
+      while (entry != null) {
+        val individualFile = entry.getName
+        //only unzip the dir containing the model
+        if (!individualFile.contains(".jar") && !individualFile.contains(".json")) {
+          // todo: asood1 create these directories and subdirectories dynamically
+          if (individualFile.contains("topicsGivenDocFrame")) {
+            fileStr = tmpDir + "/topicsGivenDocFrame"
+          }
+          else if (individualFile.contains("distLdaModel/data/globalTopicTotals")) {
```

Comment:
yeah...I have the todo to make this generic. I had looked briefly into it..but will do this soon...hopefully tomorrow